### DISCUSSION TITLE
Using local storage in _document.tsx

### DISCUSSION THREAD

**Discussion by teremy** [vercel/next.js#46039] [Help] [answered]
Hi. I have a custom _document.tsx file and I want my <html>-element to have a css-class assigned with the name of the current theme, that is located in local storage. 
So similar to this: 
...` 
   return (
      <Html className={theme}>
...
`
The problem seems to be, that the client's local storage isn't accessed but rather any logic is done server side. I tried to put my logic inside useEffect but it doesn't work either. Any ideas how to access client's local storage from _document.tsx?

**Comment by icyJoseph** [comment]
Hi,

Unfortunately, for your use case, `_document` runs server side exclusively, so you can't gain access to any client side APIs at all.

What you can use instead is a cookie! 

- Checkout this talk: https://www.youtube.com/watch?v=UhLWAQdsgBY
- Repository: https://github.com/NDAify/ndaify-web/blob/master/src/pages/_document.js 
  - You can work your way from their `_document` example I believe

**Comment by teremy** [comment, marked as answer]
Thanks for your answer. What I did now was to wrap everything in a div-container (the first child of the body-element) and assign a class to that container (either light or dark). I did this in _app.tsx, so having the logic run client-side is no longer a problem. 
It's probably possible to access the html or body element within _app.tsx, but this was a fast workaround and it does just fine. 
It was important to me, that I can use local storage and not use a cookie for this.
